src/app/account/new-order/new-order.component.ts

```diff
@@ -475,32 +477,31 @@ export class NewOrderComponent implements OnInit, OnDestroy {
           this.apiOrderService.deleteContact(invoiceContact.Id).subscribe(confirmed => {
             if (confirmed) {
               invoiceContact.Id = -1;
-              this._createOrUpdateDraftOrder(invoiceContact);
+              this._createOrUpdateDraftOrder(invoiceContact, page);
             }
           });
         }
         dialogRef = null;
       });
     } else {
-      this._createOrUpdateDraftOrder(invoiceContact);
+      this._createOrUpdateDraftOrder(invoiceContact, page);
     }
   }
 
-  private _createOrUpdateDraftOrder(invoiceContact: Contact | undefined) {
+  private _createOrUpdateDraftOrder(invoiceContact: Contact | undefined, page: number = 0) {
     this.currentOrder.title = this.orderFormGroup.get('title')?.value;
     this.currentOrder.invoice_reference = this.orderFormGroup.get('invoice_reference')?.value;
     this.currentOrder.email_deliver = this.orderFormGroup.get('emailDeliver')?.value;
     this.currentOrder.description = this.orderFormGroup.get('description')?.value;
     this.currentOrder.order_type = this.orderFormGroup.get('orderType')?.value.name;
-
     if (this.currentOrder.id === -1) {
       this.currentOrder.invoiceContact = invoiceContact;
       this.apiOrderService.createOrder(this.currentOrder.toPostAsJson, invoiceContact, this.IsAddressForCurrentUser)
         .subscribe(newOrder => {
           if (newOrder) {
             this.resetCustomerForm();
             this.storeService.addOrderToStore(new Order(newOrder as IOrder));
-            this.stepper.next();
+            this.stepper.selectedIndex = page;
```

Comment:
Is it safe to never call "next()" anymore?
It means the caller of this function must always provide a meaningful "page", right? I'm not sure that's the case in the new version of the code.